cmd/livepeer_cli/wizard_ticketbroker.go

```diff
@@ -0,0 +1,217 @@
+	fmt.Printf("Current Deposit: %v\n", currDeposit)
+	fmt.Printf("Current Penalty Escrow: %v\n", currPenaltyEscrow)
+
+	fmt.Printf("Enter deposit amount in ETH (%v will be used cover the minimum penalty escrow) - ", eth.FormatUnits(reqPenaltyEscrowAmount, "ETH"))
+
+	amount := w.readFloatAndValidate(func(in float64) (float64, error) {
+		amountWei := eth.ToBaseUnit(big.NewFloat(in))
+
+		if amountWei.Cmp(reqPenaltyEscrowAmount) < 0 {
+			return 0, fmt.Errorf("must deposit at least %v to cover the minimum penalty escrow", eth.FormatUnits(reqPenaltyEscrowAmount, "ETH"))
+		}
+
+		return in, nil
+	})
+
+	form := url.Values{
+		"amount": {eth.ToBaseUnit(big.NewFloat(amount)).String()},
+	}
+	fmt.Println(httpPostWithParams(fmt.Sprintf("http://%v:%v/fundAndApproveSigners", w.host, w.httpPort), form))
+}
+
+func (w *wizard) fundDeposit(currDeposit *big.Int) {
+	fmt.Printf("Current Deposit: %v\n", currDeposit)
+
+	fmt.Printf("Enter deposit amount in ETH - ")
+
+	amount := w.readFloat()
+
+	form := url.Values{
+		"amount": {eth.ToBaseUnit(big.NewFloat(amount)).String()},
+	}
+	fmt.Println(httpPostWithParams(fmt.Sprintf("http://%v:%v/fundDeposit", w.host, w.httpPort), form))
+}
+
+func (w *wizard) unlock() {
+	sender, err := w.senderInfo()
+	if err != nil {
+		glog.Errorf("Error getting sender info: %v", err)
+		return
+	}
+
+	if sender.WithdrawBlock.Cmp(big.NewInt(0)) > 0 {
+		fmt.Println("Already in the unlock period")
+		return
+	}
+
+	fmt.Printf("Current Deposit: %v\n", eth.FormatUnits(sender.Deposit, "ETH"))
+	fmt.Printf("Current Penalty Escrow: %v\n", eth.FormatUnits(sender.PenaltyEscrow, "ETH"))
+
+	fmt.Printf("Would you like initiate the unlock period? (y/n) - ")
+
+	input := w.readStringAndValidate(func(in string) (string, error) {
+		if in != "y" && in != "n" {
+			return "", errors.New("Enter (y)es or (n)o")
+		}
+
+		return in, nil
+	})
+	if input == "n" {
+		return
+	}
+
+	fmt.Println(httpPost(fmt.Sprintf("http://%v:%v/unlock", w.host, w.httpPort)))
+}
+
+func (w *wizard) cancelUnlock() {
+	sender, err := w.senderInfo()
+	if err != nil {
+		glog.Errorf("Error getting sender info: %v", err)
+		return
+	}
+
+	if sender.WithdrawBlock.Cmp(big.NewInt(0)) <= 0 {
+		fmt.Println("Unlock period has not been initiated")
+		return
+	}
+
+	fmt.Printf("Withdraw Block: %v\n", sender.WithdrawBlock)
+
+	fmt.Printf("Would you like to cancel the unlock period? (y/n) - ")
+
+	input := w.readStringAndValidate(func(in string) (string, error) {
+		if in != "y" && in != "n" {
+			return "", errors.New("Enter (y)es or (n)o")
+		}
+
+		return in, nil
+	})
+	if input == "n" {
+		return
+	}
+
+	fmt.Println(httpPost(fmt.Sprintf("http://%v:%v/cancelUnlock", w.host, w.httpPort)))
+}
+
+func (w *wizard) withdraw() {
+	sender, err := w.senderInfo()
+	if err != nil {
+		glog.Errorf("Error getting sender info: %v", err)
+		return
+	}
+
+	if sender.WithdrawBlock.Cmp(big.NewInt(0)) <= 0 {
```

Comment:
Just to make sure - are we making an explicit decision to not defend users from trying to withdraw before the withdraw block has passed? IIUC this code will result in gas spend even before that block has passed.

wdyt?

Reply:
Updated to prevent users from withdrawing if the sender is not unlocked [here](https://github.com/livepeer/go-livepeer/pull/658/files#diff-282e3c4a46af4e072fc859cc195b88eeR203). 

Note: If the node is using gas estimation (instead of selecting fixed gas limit), technically the user wouldn't have wasted gas because gas estimation would fail and the tx would not be submitted.